agent/xds/config.go

```diff
@@ -67,36 +67,67 @@ func ParseProxyConfig(m map[string]interface{}) (ProxyConfig, error) {
 type MeshGatewayConfig struct {
+	BindTaggedAddresses bool                              `mapstructure:"envoy_mesh_gateway_bind_tagged_addresses"`
+	BindAddresses       map[string]structs.ServiceAddress `mapstructure:"envoy_mesh_gateway_bind_addresses"`
+	NoDefaultBind       bool                              `mapstructure:"envoy_mesh_gateway_no_default_bind"`
+	ConnectTimeoutMs    int                               `mapstructure:"connect_timeout_ms"`
+}
+
+type GatewayConfig struct {
 	// BindTaggedAddresses when set will cause all of the services tagged
 	// addresses to have listeners bound to them in addition to the main service
 	// address listener. This is only suitable when the tagged addresses are IP
 	// addresses of network interfaces Envoy can see. i.e. When using DNS names
 	// for those addresses or where an external entity maps that IP to the Envoy
 	// (like AWS EC2 mapping a public IP to the private interface) then this
 	// cannot be used. See the BindAddresses config instead
-	//
-	// TODO - wow this is a verbose setting name. Maybe shorten this
-	BindTaggedAddresses bool `mapstructure:"envoy_mesh_gateway_bind_tagged_addresses"`
+	BindTaggedAddresses bool `mapstructure:"envoy_gateway_bind_tagged_addresses"`
 
 	// BindAddresses additional bind addresses to configure listeners for
-	BindAddresses map[string]structs.ServiceAddress `mapstructure:"envoy_mesh_gateway_bind_addresses"`
+	BindAddresses map[string]structs.ServiceAddress `mapstructure:"envoy_gateway_bind_addresses"`
 
 	// NoDefaultBind indicates that we should not bind to the default address of the
 	// gateway service
-	NoDefaultBind bool `mapstructure:"envoy_mesh_gateway_no_default_bind"`
+	NoDefaultBind bool `mapstructure:"envoy_gateway_no_default_bind"`
 
 	// ConnectTimeoutMs is the number of milliseconds to timeout making a new
 	// connection to this upstream. Defaults to 5000 (5 seconds) if not set.
 	ConnectTimeoutMs int `mapstructure:"connect_timeout_ms"`
 }
 
-// ParseMeshGatewayConfig returns the MeshGatewayConfig parsed from an opaque map. If an
+// ParseGatewayConfig returns the GatewayConfig parsed from an opaque map. If an
 // error occurs during parsing, it is returned along with the default config. This
 // allows the caller to choose whether and how to report the error
-func ParseMeshGatewayConfig(m map[string]interface{}) (MeshGatewayConfig, error) {
-	var cfg MeshGatewayConfig
-	err := mapstructure.WeakDecode(m, &cfg)
+func ParseGatewayConfig(m map[string]interface{}) (GatewayConfig, error) {
+	var meshCfg MeshGatewayConfig
+
+	// Decode deprecated mesh-gateway fields into Gateway Config
+	err := mapstructure.WeakDecode(m, &meshCfg)
+	cfg := GatewayConfig{
+		BindTaggedAddresses: meshCfg.BindTaggedAddresses,
+		BindAddresses:       meshCfg.BindAddresses,
+		NoDefaultBind:       meshCfg.NoDefaultBind,
+		ConnectTimeoutMs:    meshCfg.ConnectTimeoutMs,
+	}
+
+	// Merge in values from new flags if they were specified
```

Comment:
Why not use `lib.TranslateKeys` to translate the old names to the new ones and follow on with the `mapstructure.WeakDecode` into the final `GatewayConfig` struct.

Then you could also get rid of the old `MeshGatewayConfig` struct.

Reply:
Ah, forgot about that function. Thanks!